distributed/tests/test_utils_comm.py

```diff
@@ -143,3 +151,13 @@ async def my_sleep(amount):
 
     assert n_calls == 6
     assert sleep_calls == [0.0, 1.0, 3.0, 6.0, 6.0]
+
+
+@pytest.mark.parametrize("dict_type", [dict, OrderedDict])
+def test_unpack_remotedata_unpacks_dicts(dict_type):
+    rd1 = WrappedKey("key-1")
+    rd2 = WrappedKey("key-2")
+    assert unpack_remotedata(dict_type([(1, rd1), (2, rd2)])) == (
+        dict_type([(1, "key-1"), (2, "key-2")]),
+        {rd1, rd2},
+    )
```

Comment:
Thanks you for adding this lower-level test for `unpack_remotedata`. Could you also add a higher-level test to `distributed/tests/test_client.py` which ensure the user-facing result from the original issue is as expected? For example, we could add an `assert` that

```python
client.submit(show, OrderedDict({"key": client.submit(identity, "hello")}))
```

(or something similar) is returning the correct result.

Reply:
👍 will do

Reply:
This file doesn't seem to have any tests of that sort (but there are a lot of tests in there). Should I just put it at the bottom or is there somewhere else in there that makes sense to you ?

Reply:
No strong preference, wherever you prefer in that module

Reply:
I added a test but it's failing for `OrderedDict` -- looks like there's somewhere else that needs changes :( good call on adding a client level test!